org.eclipse.winery.frontends/app/tosca-management/src/app/instance/sharedComponents/parameters/parameters.component.ts

```diff
@@ -40,52 +43,89 @@ export class ParametersComponent {
     ];
     @Input() enableFiltering = false;
 
+    @Output() onParameterEdited = new EventEmitter<Parameter>();
     @Output() onParameterAdded = new EventEmitter<Parameter>();
     @Output() onParameterRemoved = new EventEmitter<Parameter>();
 
-    @ViewChild('modal') modal: ModalDirective;
-    @ViewChild('confirmRemoveModal') confirmRemoveModal: ModalDirective;
+    dynamicTableData: Array<WineryDynamicTableMetadata> = [];
 
     validatorObject: WineryValidatorObject;
 
-    param: Parameter = new Parameter();
-    selectedParam: Parameter;
-
-    constructor(public instanceService: InstanceService) {
-    }
-
-    openModal() {
-        this.param = new Parameter();
-        this.validatorObject = new WineryValidatorObject(this.parameters, 'key');
-        this.modal.show();
+    ngOnInit(): void {
+        this.dynamicTableData = [
+            new DynamicTextData(
+                'key',
+                'Name',
+                0,
+                [Validators.required],
+            )];
+        if (this.containsColumn('type')) {
+            this.dynamicTableData.push(
+                new DynamicDropdownData<'string' | 'integer' | 'float' | 'boolean' | 'timestamp'>(
```

Comment:
Why don't you use the `YamlType` defined in the propertiesDefinition.component.ts? Maybe think about moving the types to a separate file.